### PR TITLE
dropped mAnkiActivity from CreateDeckDialog

### LINKED ISSUE #9161: [Bug] 2.15.4: CreateDeckDialog.createNewDeck - NullPointerException
From Google Play. Number 1 crash

```
java.lang.NullPointerException: 
  at android.content.ContextWrapper.getPackageName (ContextWrapper.java:142)
  at android.preference.PreferenceManager.getDefaultSharedPreferencesName (PreferenceManager.java:537)
  at android.preference.PreferenceManager.getDefaultSharedPreferences (PreferenceManager.java:526)
  at com.ichi2.anki.AnkiDroidApp.getSharedPrefs (AnkiDroidApp.java:371)
  at com.ichi2.anki.CollectionHelper.getCurrentAnkiDroidDirectory (CollectionHelper.java:263)
  at com.ichi2.anki.CollectionHelper.getCollectionPath (CollectionHelper.java:255)
  at com.ichi2.anki.CollectionHelper.getCol (CollectionHelper.java:122)
  at com.ichi2.anki.CollectionHelper.getCol (CollectionHelper.java:115)
  at com.ichi2.anki.AnkiActivity.getCol (AnkiActivity.java:157)
  at com.ichi2.anki.dialogs.CreateDeckDialog.createNewDeck (CreateDeckDialog.java:139)
  at com.ichi2.anki.dialogs.CreateDeckDialog.createDeck (CreateDeckDialog.java:114)
  at com.ichi2.anki.dialogs.CreateDeckDialog.onPositiveButtonClicked (CreateDeckDialog.java:153)
  at com.ichi2.anki.dialogs.CreateDeckDialog.lambda$showDialog$0 (CreateDeckDialog.java:99)
  at com.ichi2.anki.dialogs.CreateDeckDialog.lambda$showDialog$0$CreateDeckDialog (CreateDeckDialog.java)
  at com.ichi2.anki.dialogs.-$$Lambda$CreateDeckDialog$GT3-9pD_4kBO9MoWULdODVWg6UI.onClick (lambda:2)
  at com.afollestad.materialdialogs.MaterialDialog.onClick (MaterialDialog.java:410)
  at android.view.View.performClick (View.java:6659)
  at android.view.View.performClickInternal (View.java:6631)
  at android.view.View.access$3100 (View.java:790)
  at android.view.View$PerformClick.run (View.java:26187)
  at android.os.Handler.handleCallback (Handler.java:907)
  at android.os.Handler.dispatchMessage (Handler.java:105)
  at android.os.Looper.loop (Looper.java:216)
  at android.app.ActivityThread.main (ActivityThread.java:7625)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:524)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:987)
```

### PR DESCRIPTION
**Pull request by Akshay0701**
Fixes #9161 
used CollectionHelper class instead AnkiActivty.,

steps to reproduce 
1 sync with menu item
2 then pull refresh and sync 
3 try to create deck

### CODE REVIEW

**Comment by Akshay0701** [comment]
i think we should now replace all `new AnkiActivty()` with `CollectionHelper.getInstance()`

**Comment by Akshay0701** [comment]
now there is only one class which using `new AnkiActivty()`  instead of `CollectionHelper.getInstance()`

can I change that code also?

**Review by mikehardy — APPROVED** [review]
@david-allison-1 this looks right to me, what do you think?
And to @Akshay0701 's question, it seems like he is correct, it should always be CollectionHelper now, to get a collection?
I would support a second commit that changed the last remaining instance

**Comment by mikehardy** [comment]
Nice! Thank you @Akshay0701

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CreateDeckDialog.java [hunk 1]
```diff
@@ -177,7 +176,7 @@ public void renameDeck(String newDeckName) {
             UIUtils.showThemedToast(mContext, mContext.getString(R.string.invalid_deck_name), false);
         } else if (!newName.equals(mPreviousDeckName)) {
             try {
-                Collection col = mAnkiActivity.getCol();
+                Collection col = CollectionHelper.getInstance().getCol(mContext).getCol();
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/CreateDeckDialog.java:179]: sorry this getCol repeated, i will fix this don't merge 😅

**Review by david-allison-1 — APPROVED** [review]
Looks great!

Shouldn't use `new AnkiActivity()`, sorry I didn't catch that. Let's solve that there as well

**Comment by Akshay0701** [comment]
@david-allison-1 it's done!

File: AnkiDroid/src/test/java/com/ichi2/anki/dialogs/CreateDeckDialogTest.java [hunk 2]
```diff
@@ -73,11 +70,15 @@ public void testCreateFilteredDeckFunction() {
...
-    public void testCreateSubDeckFunction() throws FilteredAncestor {
-        Long deckParentId = new AnkiActivity().getCol().getDecks().id("Deck Name");
-
+    public void testCreateSubDeckFunction() {
         mActivityScenario.onActivity(activity -> {
-            CreateDeckDialog createDeckDialog = new CreateDeckDialog(activity, R.string.new_deck, CreateDeckDialog.DeckDialogType.SUB_DECK, deckParentId);
+            AtomicReference<Long> deckParentId = new AtomicReference<>();
+            try {
+                deckParentId.set(CollectionHelper.getInstance().getCol(activity).getDecks().id("Deck Name"));
+            } catch (FilteredAncestor filteredAncestor) {
```
Comment (david-allison-1) [inline AnkiDroid/src/test/java/com/ichi2/anki/dialogs/CreateDeckDialogTest.java:78]: Is it possible to reduce the complexity of this to how it was before

* keep a `Long`
* Use `throws`

**Comment by david-allison-1** [comment]
test-based nit, good to go regardless of answer, but it's best to keep the code simple

File: AnkiDroid/src/test/java/com/ichi2/anki/dialogs/CreateDeckDialogTest.java (hunk 2 above)
Comment (Akshay0701) [reply AnkiDroid/src/test/java/com/ichi2/anki/dialogs/CreateDeckDialogTest.java:78]: Capture

yes at very starting i wrote simple code but then it showed error (add try/catch block) then i add try, and then again error occur in method getSubdeckName() that "deckparentId should be final", and then again i took whole code in one try/catch but again error came 😭  

then I tried AtomicReference and it worked properly. 

I will love it if you could edit this code, please?

Comment (david-allison-1) [reply AnkiDroid/src/test/java/com/ichi2/anki/dialogs/CreateDeckDialogTest.java:78]: Move the first line above the lambda

Comment (david-allison-1) [reply AnkiDroid/src/test/java/com/ichi2/anki/dialogs/CreateDeckDialogTest.java:78]: Move it up, outside the lambda 

`RobolectricTest` has a `getCol()` method

**Review by david-allison-1 — APPROVED** [review]
Beauty!

Pending CI

**Comment by Akshay0701** [comment]
Thanks!

**Comment by mikehardy** [comment]
unit tests were very unhappy there, did not appear to be a normal flake but I did not investigate deeply. Restarted them, if they fail again, needs investigation

**Comment by Akshay0701** [comment]
15 tests fails.

**Comment by mikehardy** [comment]
I'll validate locally, David and I think this is something unrelated

**Comment by mikehardy** [comment]
Verified locally with API29/API30 emulator  `./gradlew clean jacocoTestReport`

**Comment by mikehardy** [comment]
release note: partial pick the first time - only one of the two commits. Re-milestoning to 2.15.6 for release with the second commit

**Comment by mikehardy** [comment]
Hi there!

Apologies for the delay, and if you're already submitted something to OpenCollective for PRs merged in June 2021 PRs merged, you may feel free to ignore this and it will be processed shortly

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for June 2021 submissions

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

(I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged in May)

Thanks!

**Pull request merged**